index.js

```diff
@@ -17,9 +17,14 @@ export default async function run() {
     // Wait for all the setup() promises to resolve
     if (runAsync) {
         core.info("Running setups in parallel")
-        const setups = Tool.all().map((tool) =>
-            tool.setup(core.getInput(tool.name)),
-        )
+        const setups = Tool.all().map(async (tool) => {
+            try {
+                await tool.setup(core.getInput(tool.name))
+            } catch (error) {
+                core.error(`caught error in ${tool.name} setup: ${error}`)
+                throw error
+            }
+        })
```

Comment:
I think maybe this changes this from setting up in parallel, as now we wait for each tool.setup to complete before going to the next. I imagine this might the reality if we want catch all errors logs their messages but did imagine we would be able to catch them in catch block on line 61.

Reply:
I think this still remains parallel. We put the error handling here so we could log what tool failed. We could refactor to a `.catch` but I think it would be the same

Reply:
Simplified example that I think confirms that things still run in parallel:

```
await Promise.all([
    (async () => await new Promise(resolve => setTimeout(() => console.log("First") && resolve(), 1000)))(), 
    (async () => await new Promise(resolve => setTimeout(() => console.log("Second") && resolve(), 500)))()
])
```

Reply:
ah... try catch async await. I definitely thought there was a chance I was mixed up on this. I think I get caught up on the fact that we await tool.setup(core.getInput(tool.name), which doesn't really wait, yet, we can still utilize the catch block here.

Reply:
I do wonder if we can log the errors on line 61 or if somehow those errors are disconnected and wouldn't get logged

Reply:
oh this PR is closed, nevermind!

Reply:
We can always create a new one with your feedback addressed. I just went ahead and merged it to unblock the failing CI jobs

Reply:
it's all good, nothing like that.  I just figured it was 'in the past' and I was still talking about it. 😆

Reply:
> I do wonder if we can log the errors on line 61 or if somehow those errors are disconnected and wouldn't get logged

I think we could add the logging there as well. Since `run` all it does is execute each tool and we are now logging things it may not be needed, but if in the future we change the code, we guarantee that we are still logging errors.

Some other improvements:

* Remove dependency in `node-version-data` and instead roll out our own inside this repo (we don't need `iojs` at all)
* Improve the way the local cache is setup (mayge have a periodic job that updates things) and trim down the list (we don't need anything older that Node 16).

Reply:
I see. Thanks for extra context!